sdccc/src/main/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditions.java

```diff
@@ -398,6 +408,29 @@ static Optional<String> associateNewLocationForHandle(
             return stateHandle;
         }
 
+        private static void waitForEpisodicContextReportToModifyStateHandle(
+                final TestClient testClient, final String stateHandle) {
+            final EpisodicContextReportStateHandleObserver observer =
+                    new EpisodicContextReportStateHandleObserver(stateHandle);
+            testClient.getSdcRemoteDevice().getMdibAccessObservable().registerObserver(observer);
+
+            final long start = System.currentTimeMillis();
+            final long end = start + TIMEOUT_EPISODIC_CONTEXT_REPORT_MILLIS;
+            long now = System.currentTimeMillis();
+            boolean done = false;
+            synchronized (observer.getSignal()) {
+                while (now < end && !done) {
+                    try {
+                        observer.waitForStateUpdate(end - now);
```

Comment:
You could just use your _TIMEOUT_EPISODIC_CONTEXT_REPORT_MILLIS_

Reply:
Done.